### PR TITLE
Update connector design to reflect platform realities

### PR DESCRIPTION
**Pull request by nharring-adjacent**
Since there are interdependencies between syntax changes for connectors and sources and making secrets work for supplying sensitive credentials in platform it is valuable to make sure the design doc is more accurate than it was with regards to what fields will have which types of values.

### Motivation

   * This PR refactors existing code.

Existing design is sufficiently out of date to cause confusion when discussing current work.

### Testing

- [X] This PR has adequate test coverage / QA involvement has been duly considered.

### CODE REVIEW

**Comment by mkysel** [comment]
can you fix up the examples starting on line 48 to follow the new syntax?

File: doc/developer/design/20220404_connectors.md [hunk 1]
```diff
@@ -124,11 +119,11 @@ FOR CONFLUENT SCHEMA REGISTRY
...
   USERNAME [=] <value>
-  PASSWORD [=] <value>
+  PASSWORD [=] <secret name>
   SSL (
-    CERTIFICATE FILE [[=] <value> ]
-    KEY FILE [[=] <value> ]
-    CA FILE [[=] <value> ] )
+    CERTIFICATE [[=] <secret name> ]
+    KEY [[=] <secret name> ]
+    CA FILE [[=] <secret name> ] )
```
Comment (mkysel) [inline doc/developer/design/20220404_connectors.md:126]: `CA FILE` or `CA`? I suspect just CA.

**Comment by mkysel** [comment]
When playing around with your branch, I noticed this:
```
materialize=> CREATE CONNECTOR kafka FOR
KAFKA BROKER 'kafka:9092' SECURITY;
ERROR:  Expected one of SSL or SASL or SASLSSL, found semicolon
```

In the spec, we only support SSL and SASL.The two grammars should probably match.

**Comment by mkysel** [comment]
Looks good to me, but I think that you should wait for @benesch to express his opinion

**Review by mkysel — APPROVED** [review]

**Review by benesch — APPROVED** [review]

File: doc/developer/design/20220404_connectors.md [hunk 2]
```diff
@@ -46,19 +46,22 @@ With the connectors proposed in this document, the `CREATE SOURCE` can instead
...
-    security_protocol = 'SASL_SSL',
-    sasl_mechanisms = 'PLAIN',
-    sasl_username = 'username',
-    sasl_password = 'password'
-);
+KAFKA BROKER 'kafka:9092'
+SECURITY SaslSsl
+MECHANISM PLAIN
+USERNAME 'username'
+PASSWORD kafka_pass;
```
[the remark this answers is not in the document]
Comment (nharring-adjacent) [reply doc/developer/design/20220404_connectors.md:56]: Yeah, good call. I will need to make more extensive changes to this based on how the code has evolved in the last couple days to get tests actually passing using secrets anyhow but I will make sure to reflect this feedback also. :)

**Comment by benesch** [comment]
We're going to revisit the syntax questions here once #5390 is fixed.

**Pull request closed without merging**